applications/asset_tracker/src/main.c

```diff
@@ -410,7 +413,12 @@ static void send_agps_request(struct k_work *work)
 
 	LOG_INF("Sending A-GPS request");
 
+#if defined(CONFIG_NRF_CLOUD_CELL_POS)
+	err = nrf_cloud_request_cell_location(CELL_LOC_TYPE_SINGLE,
+		(bool)IS_ENABLED(CONFIG_NRF_CLOUD_CELL_POS_REQ_CELL_BASED_LOC));
+#else
 	err = gps_agps_request(agps_request, GPS_SOCKET_NOT_PROVIDED);
+#endif
 	if (err) {
 		LOG_ERR("Failed to request A-GPS data, error: %d", err);
```

Comment:
The name of this function is `send_agps_request()` but it's doing two different things. It would be clearer to split it in two functions, one for sending an AGPS data request and one for sending a cellular positioning request. Currently also the log output always talks about AGPS even when cellular positioning is used.

Reply:
The code use the agps Kwork to fetch cell_pos or request agps. Just tried to make minimal change to the code. If a better solution is required, will leave this to assert_tracker authors.